### ISSUE TITLE
Open to nested action

### ISSUE THREAD

**Issue by TNorbury** [TNorbury/command_palette#4]
Extension of the programmatic controls. This would open up the command palette and set a given (nested) action as the selected one.

For example, say we have a button to change the theme, upon pressing that the command palette would open, with the "Change Theme" action already selected, and thus showing the different theme options.

My biggest question is how will we specify the action. First thought is passing an Action object that will be matched against, however this would require the user to always have a reference to it (not the end of the world, but could be cumbersome). Another option would be to allow IDs to be assigned to each Action, then the user could just open via ID.

Another question is: if the palette was opened with an action, should we still preserve the backspace to go up behavior, or should this be a case where a backspace will close the palette? Perhaps this will be best left as a config option?

- [ ] Open command palette with given action selected.
- [ ] Open with Action object
- [ ] Open with ID
- [ ] Backspace behavior (up or exit?)

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Thoughts on what to do after thinking on it. I'll go the ID route, and backspace behavior here won't go "up" or close the palette. It'll just delete characters like normal

**Comment by bit-garden** [comment]
Yes on the ID bit, but maybe expose it to the user too. Like `#` or `>theme` would display the theme commands. You can add an argument to open like this `CommandPalette.of(context).open([String textPrefix])`. Then the user can just delete the prefix if they decide against it. 

With this, you wouldn't be ambiguously dropping the user into a tree of options somewhere, but still getting them the direct options they need without much extra code for the developer(I'm thinking just an extra argument for the CommandPaletteAction). This would also hint to the user another way to jump around the nested palette actions faster without needing the UI.

**Comment by caleb654** [comment]
This would be sweet. Even just being able to open to a particular nested option programmatically via our own custom keyboard shortcuts I think would work well.

**Comment by Arjun2908** [comment]
I want to have this feature in my app, any updates on when this is going to be added?

**Comment by TNorbury** [comment]
@Arjun2908 I have the functionality implemented, I'm writing a test for it currently, but it has been on the back burner as I've been busy with other things. I'll plan to have it out by the end of the month

**Comment by TNorbury** [comment]
This feature is included in v0.5.0 which has been published to pub 🥳

**Comment by caleb654** [comment]
> This feature is included in v0.5.0 which has been published to pub 🥳

Awesome thank you for getting this feature out so quickly!!
